### PR TITLE
<WIP>: Same build of OE SDK for both FLC/non FLC machines

### LINKED ISSUE #3178: FLC/Non-FLC in one build?
OE has `HAS_QUOTE_PROVIDER=ON|OFF` as build-time option. Since the Intel SGX libraries are now all dlopen'd I was wondering whether it makes sense to allow building a single OE variant that works across all SGX variants (currently FLC/non-FLC) with appropriate runtime introspection.

### PR DESCRIPTION
**Pull request by anakrish**
This allows OE SDK packages to be built on non FLC machines.

Signed-off-by: Anand Krishnamoorthi <anakrish@microsoft.com>
fixes #3178

### CODE REVIEW

**Comment by anakrish** [comment]
bors try

File: include/openenclave/bits/sgx/sgxtypes.h [hunk 1]
```diff
@@ -1035,6 +1035,17 @@ typedef struct _sgx_key
 #define OE_SEALKEY_DEFAULT_MISCMASK (~SGX_MISC_NON_SECURITY_BITS)
 #define OE_SEALKEY_DEFAULT_XFRMMASK (0X0ULL)
 
+#define SGX_QL_MK_ERROR(x) (0x0000E000 | (x))
+
+// clang-format off
+/** Errors generated by the quote interface that OE cares about. */
+typedef enum _quote3_error_t {
```
Comment (radhikaj) [inline include/openenclave/bits/sgx/sgxtypes.h:1042]: How did you define what error codes OE would care about?

**Comment by anakrish** [comment]
bors try

**Comment by anakrish** [comment]
bors try-

**Comment by anakrish** [comment]
bors try

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -29,6 +29,35 @@ if (OE_SGX)
   set(DEFINE_OE_SGX "-DOE_SGX")
 endif ()
 
+# Detect if SGX_DCAP_QL is available on the machine.
+if (OE_SGX)
+  # Detect if sgx_dcap_ql is available on the machine.
```
Comment (dthaler) [inline tests/CMakeLists.txt:34]: isn't this comment redundant with line 32?

**Comment by radhikaj** [comment]
@anakrish Can the conflicts be resolved?
@BRMcLaren Can this be merged?

**Comment by anakrish** [comment]
All changes from this experimental PR has been merged in via individual PRs. This PR is no longer relevant.

**Pull request closed without merging**